psydac/feec/derivatives.py

```diff
@@ -169,7 +169,7 @@ def dot(self, v, out=None):
 
         # setup, space checks
         assert v.space is self._domain
-        v.update_ghost_regions(direction=self._diffdir)
+        assert v.ghost_regions_in_sync
```

Comment:
Here we should update the ghost regions if we are in 1D (please add a comment explaining why)

Reply:
done in [1ba0b57](https://github.com/pyccel/psydac/pull/240/commits/1ba0b571463f10a05691ba6e80d953ac445c157c)